src/ResourceManager/Network/Commands.Network/VirtualNetwork/Subnet/Delegation/NewAzureDelegationCommand.cs

```diff
@@ -0,0 +1,47 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network.VirtualNetwork.Subnet
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "Delegation", DefaultParameterSetName = "SetByResource"), OutputType(typeof(PSDelegation))]
```

Comment:
what is the use case for this cmdlet?
[Add](https://github.com/Azure/azure-powershell/pull/7220/files#diff-cbe5efc62d84536dab8297a645180adfR22) does not take in a delegation object.

Whenwill the cmdlet be used?

Reply:
It can be passed in with operations on the subnet
New-AzureRmVirtualNetworkSubnetConfig, Add-AzureRmVirtualNetworkSubnetConfig